ygen/enumgen.go

```diff
@@ -71,6 +71,67 @@ func newEnumSet() *enumSet {
 	}
 }
 
```

Comment:
Throughout this code I wonder whether "duplicate" is the right wording. Particularly because I think there are two cases that I can see:

```yang

grouping A {
  leaf b { type enumeration { enum C; } }
}

container one {
  uses A;
}

container two {
  uses B;
}
```

in this case, leaf `one/b` and `two/b` are output once in our current approach (without `skipDedup` enabled). However, I think your duplicate sets issue is more dealing with the case where we have two values that come up with the same `E_Foo_Bar` type, right?

Can we call these `NameClash` sets or something to be clearer?